luma/core/virtual.py

```diff
@@ -227,8 +229,14 @@ def println(self, text=""):
         Prints the supplied text to the device, scrolling where necessary.
         The text is always followed by a newline.
         """
-        self.puts(text)
-        self.newline()
+        if self.word_wrap:
+            text = wrap(text, width=self.width)
```

Comment:
I want to look into the issue a bit more before merging, thanks for the feedback.